src/cli_arguments.h

```diff
@@ -0,0 +1,77 @@
+#ifndef __CLI_ARGUMENTS_H_INCLUDED__
+#define __CLI_ARGUMENTS_H_INCLUDED__
+
+#include<iostream>
+
+using namespace std;
+class CLIarguments {
+
+	int dout;
+	int din;
+	int pen;
+
+public:
+
+	bool SetDout(int dout) { this->dout = dout; return 1; }
+	bool SetDin(int din) { this->din = din; return 1; }
+	bool SetPen(int pen) { this->pen = pen; return 1; }
+	int GetDout() { return this->dout; }
+	int GetDin() { return this->din; }
+	int GetPen() { return this->pen; }
+
+	CLIarguments(int dout = 0, int din = 0, int pen = 0) { //def constructor
+		this->din = din;
+		this->dout = dout;
+		this->pen = pen;
+	}
+	// --dout=120
+	void Parsing(char *arg) {
+		int size = strlen(arg) + 1;
+		char *buff = new char[size];
+		int j = 0, a = 0, p = 0;  //iterators or counters
```

Comment:
Semantic (human-readable) names would help:

- `buff` &rarr; `strippedArgument` or `argumentWithoutControlChars` or `alphaNumericChars` or something like that;
- `j` &rarr; `%prevous_variable_name%Index`;
- `a` &rarr; `numberPartIndex`;
- `p` &rarr; `namePartIndex`.

Your names might be different of course but please keep readability for human in mind: why `numberPart[a]` but not `numberPart[p]`? What's the difference between `a` and `p`?